libevmasm/Assembly.cpp

```diff
@@ -568,11 +568,14 @@ std::pair<std::shared_ptr<Assembly>, std::vector<std::string>> Assembly::fromJSO
 			Json::Value const& code = data[dataItemID];
 			if (code.isString())
 			{
-				bytes data_value{fromHex(code.asString())};
-				solRequire(
-					!data_value.empty(),
-					AssemblyImportException,
-					"Member '.data' contains a value for '" + dataItemID + "' that is not a valid hexadecimal string.");
+				if (!code.asString().empty())
```

Comment:
Looks like we also allow empty hexadecimal strings, e.g. `test/libyul/yulSyntaxTests/objects/data.yul`:
```
object "A" {
  code {
    datacopy(0, dataoffset("3"), datasize("3"))
  }
  data "1" ""
  data "2" hex"0011"
  data "3" "hello world this is longer than 32 bytes and should still work"
}
```